### PR TITLE
Adds Trace Container composable

### PR DESCRIPTION
**Pull request by eri9000**
Related to issue: kotlin#4867

### Summary of changes:
- Refactors `Trace` to separate UI from Navigation 

### Pre-merge Checklist

- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/) Job for this PR has been run
  - [ ] link: https://runtime-kotlin.esri.com/view/vTest/job/vtest/job/toolkit/178/
- Unit and/or integration tests have been added to exercise this PR's logic, and the tests are passing:
  - [ ] Yes
  - [ ] No

### CODE REVIEW

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/Trace.kt [hunk 1]
```diff
@@ -49,8 +49,8 @@ import com.arcgismaps.toolkit.utilitynetworks.internal.util.TabRow
 import com.arcgismaps.toolkit.utilitynetworks.ui.TraceNavHost
 
 /**
- * A composable UI component to set up and run a [com.arcgismaps.utilitynetworks.UtilityNetwork.trace]
```
Comment (eri9000) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/Trace.kt:52]: this import was not working

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/Trace.kt [hunk 2]
```diff
@@ -61,70 +61,97 @@ public fun Trace(
...
-                                Icons.Default.Info,
-                                contentDescription = stringResource(id = R.string.error),
-                                tint = MaterialTheme.colorScheme.error
-                            )
-                            Spacer(modifier = Modifier.size(8.dp))
-                            Text(errorMessage, color = MaterialTheme.colorScheme.error)
-                        }
-                    }
-                    if (traceState.isTaskInProgress.value) {
+                is InitializationStatus.FailedToInitialize -> {
```
Comment (eri9000) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/Trace.kt:125]: this additional case will only show an unrecoverable error, not the TraceOptions screen

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/Trace.kt [hunk 3]
```diff
@@ -61,70 +61,97 @@ public fun Trace(
...
+                    selectedTabIndex,
+                    onNavigateTo = {
+                        selectedTabIndex = it.first
+                        traceState.showScreen(it.second)
+                    }
+                )
+            }
+        } else {
+            null
+        }
```
Comment (sorenoid) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/Trace.kt:86]: This seems out of place now. Shouldn't it be entirely within the `TraceContainer`? Is there a reason it must be passed as a parameter?

**Review by sorenoid — CHANGES_REQUESTED** [review]
One question

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/Trace.kt (hunk 3 above)
Comment (eri9000) [reply toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/Trace.kt:86]: I am thinking of `TraceContainer` as a `Scaffold`, where the `TopNavBar` is passed in. 
This also allows us to test the content independently from the tabRow and might make it easier to refactor it later without introducing other dependencies to `TraceContainer`.

Comment (sorenoid) [reply toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/Trace.kt:86]: ok. should it be called `TraceScaffold` then?

**Review by sorenoid — APPROVED** [review]

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/Trace.kt [hunk 4]
```diff
@@ -49,8 +49,8 @@ import com.arcgismaps.toolkit.utilitynetworks.internal.util.TabRow
 import com.arcgismaps.toolkit.utilitynetworks.ui.TraceNavHost
 
 /**
- * A composable UI component to set up and run a [com.arcgismaps.utilitynetworks.UtilityNetwork.trace]
- * on a [com.arcgismaps.toolkit.geoviewcompose.MapView].
+ * A composable UI component to set up and run a [trace](https://developers.arcgis.com/kotlin/api-reference/arcgis-maps-kotlin/com.arcgismaps.utilitynetworks/-utility-network/trace.html?)
```
Comment (puneet-pdx) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/Trace.kt:52]:
```suggestion
 * A composable UI component to set up and run a [trace](https://developers.arcgis.com/kotlin/api-reference/arcgis-maps-kotlin/com.arcgismaps.utilitynetworks/-utility-network/trace.html)
```

**Comment by puneet-pdx** [comment]
@eri9000 this refactor has changed how the circular progress indicator is being displayed. Before these changes it was in the middle of the component now it is at the top. Can we go back to how it was before.

**Before** 
https://github.com/user-attachments/assets/090978ba-5e7e-416f-9a1e-c6b64d5f3be8

**After**
https://github.com/user-attachments/assets/a75fe4d4-5912-4ce0-9a7f-7f610806de83

**Review by puneet-pdx — COMMENTED** [review]
@eri9000 We have a behavior change with this refactor.

**Review by puneet-pdx — APPROVED** [review]

**Pull request merged**